src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs

```diff
@@ -504,8 +503,15 @@ private async Task StopAsyncCore(bool disposing)
 
                 if (disposing)
                 {
-                    (_serviceProvider as IDisposable)?.Dispose();
                     _disposed = true;
+                    if (_serviceProvider is IAsyncDisposable asyncDispose)
```

Comment:
This is fun, `_serviceProvider` calls `HubConnection.DisposeAsync` so we need to set the bool first.

And for netstandard2.0 we need to keep `IDisposable` I believe.

Reply:
Add a comment about setting the bool in the correct order.